server.js

```diff
@@ -23,16 +39,112 @@ mongoose.Promise = Promise
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+// MIDDLEWARES
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+// CHECKS IF SERVICE IS AVAILABLE
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1 || 2) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
+
+// ROUTES
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Welcome to the Golden Globes API. Possible endpoints: /nominations (possible queries: category, nominee and film), /nominatons/id/:id, /nominations/:year_award (possible query: win=true), /nominations/category/:category (possible query: win=true), /nominations/category/:category/year/:year (possible query: win=true)')
+})
+
+//SHOWS ALL THE NOMINATIONS
+app.get('/nominations', async (req, res) => {
+  const { category, nominee, film } = req.query
+
+  // POSSIBLE QUERIES: CATEGORY, NOMINEE & FILM
+  const showNominations = await Nomination.find({
+    category: new RegExp(category, 'i'),
+    nominee: new RegExp(nominee, 'i'),
+    film: new RegExp(film, 'i'),
+  })
+
+  if (showNominations) {
+    res.json(showNominations)
+  } else {
+    res.status(404).json({ error: 'Error in search' })
+  }
+})
+
+// SHOW NOMINATION BY ID
+app.get('/nominations/id/:id', async (req, res) => {
+  const nominations = await Nomination.findById(req.params.id)
+  res.json(nominations)
+})
+
+// SHOW ALL NOMINATIONS IN A YEAR
+app.get('/nominations/:year_award', async (req, res) => {
+  const { year_award } = req.params
+  let yearNominations = await Nomination.find({ year_award: year_award })
+
+  // FILTER THE NOMINATIONS IN A YEAR TO ONLY SHOW THE ONES WHO WON
+  const showWin = req.query.win
+
+  if (showWin && yearNominations) {
+    yearNominations = yearNominations.filter((item) => item.win === true)
+  }
+  res.json(yearNominations)
+
+  if (yearNominations) {
+    res.json(yearNominations)
+  } else {
+    res.status(404).json({ error: 'Not found' })
+  }
+})
```

Comment:
Here you are fetching a big set of data no matter if you want only the winners or not.
I think a better option would be to check if win=true and then use `Nomination.find()` with or without win parameter to not load your servers with extra computing and transfers?